doc/source/overview/config_files.md

```diff
@@ -0,0 +1,753 @@
+[preprocess_fn]
+src = "function.dill"
+```
+
+Alternatively, the following are also equivalent:
+
+<p class="codeblock-label">config.toml (excerpt)</p>
+
+```toml
+preprocess_fn = {src="function.dill"}
+```
+
+<p class="codeblock-label">config.toml (excerpt)</p>
+
+```toml
+preprocess_fn.src = "function.dill"
+```
+
+The above schemas are not strictly necessary in this case, since `preprocess_fn` could simply be specified with
+`preprocess_fn = "function.dill"` here. However, the need to specify fields as dictionaries will become clear in the 
+following sections, where the layouts of the various possible artefact dictionaries are documented.
+
+#### Preprocessing function
+
+Simple `preprocess_fn`'s can be specified directly as a serialized dill file or via a function registry. If additional 
+arguments are to be passed at call time, the `preprocess_fn` can instead be specified as a dictionary:
+
+<p class="codeblock-label">config.toml (excerpt)</p>
+
+```toml
+[preprocess_fn]
+src = "function.dill"
+kwargs = {kwarg1="a", kwarg2=1.5, kwarg3=true}
+```
+
+In this case, the `src` field can be a locally stored `.dill` file or a function registry, and `kwargs` is a 
+dictionary of keyword arguments to be passed to the function. As noted previously, there is considerable flexibility 
+with regard to how these fields are specified. For example, the following is equivalent:
+
+<p class="codeblock-label">config.toml (excerpt)</p>
+
+```toml
+preprocess_fn.src = "function.dill"
+
+[preprocess_fn.kwargs]
+kwarg1 = "a"
+kwarg2 = 1.5
+kwarg3 = true
+```
+
+A special (but common!) case for `preprocess_fn` is to use the Alibi Detect utility function 
+[preprocess_drift](../api/alibi_detect.cd.tensorflow.preprocess.rst) (or its pytorch equivalent). This is a utility 
+function to incorporate a user-supplied model, batch preprocessing function, and/or tokenizer into the preprocessing 
+step of a drift detector. It is pre-registered, so can be specified with 
+`src="@cd.tensorflow.preprocess.preprocess_drift"`, and the rest of the fields in `preprocess_fn` are then the 
+args/kwargs to pass to `preprocess_drift`. For example: 
+
+<p class="codeblock-label">config.toml (excerpt)</p>
+
+```toml
+[preprocess_fn]
+src = "@cd.tensorflow.preprocess.preprocess_drift"
+batch_size = 32
+preprocess_batch_fn = "batch_fn.dill"
+```
+
+#### Models 
+
+A `model` dictionary may be used to specify a model within a `preprocess_fn`, or a model for detectors such as the
+[ClassifierDrift](../api/alibi_detect.cd.classifier.rst) detector. The possible fields are:
+
+```{list-table} Model schema
+:header-rows: 1
+:name: model-table
+:widths: auto
+
+* - Field
+  - Description
+  - Default (if optional)
+
+* - src
+  - Filepath to directory storing the model (relative to the `config.toml` file, or absolute). 
+  -
+  
+* - layer
+  - Optional index of hidden layer to extract. If not `None`, a 
+  [HiddenOutput](../api/alibi_detect.cd.tensorflow.preprocess.rst) model is returned.
```

Comment:
Maybe this is one for docstrings too, but I'm not sure what it means for a `HiddenOutput` model to be returned?

Reply:
Thanks, will do.